server/jwt_test.go

```diff
@@ -3180,8 +3187,228 @@ func TestAccountNATSResolverFetch(t *testing.T) {
+		uclaim.Expires = expiration.Unix()
+	}
+	vr := jwt.ValidationResults{}
+	uclaim.Validate(&vr)
+	require_Len(t, len(vr.Errors()), 0)
+	ujwt, err := uclaim.Encode(accKp)
+	require_NoError(t, err)
+	return genCredsFile(t, ujwt, seed)
+}
+
+func TestJWTUserLimits(t *testing.T) {
+	// helper for time
+	inAnHour := time.Now().Add(time.Hour)
+	inTwoHours := time.Now().Add(2 * time.Hour)
+	doNotExpire := time.Now().AddDate(1, 0, 0)
+	// create account
+	kp, _ := nkeys.CreateAccount()
+	aPub, _ := kp.PublicKey()
+	claim := jwt.NewAccountClaims(aPub)
+	aJwt, err := claim.Encode(oKp)
+	require_NoError(t, err)
+	conf := createConfFile(t, []byte(fmt.Sprintf(`
+		listen: -1
+		operator: %s
+		resolver: MEM
+		resolver_preload: {
+			%s: %s
+		}
+    `, ojwt, aPub, aJwt)))
+	defer os.Remove(conf)
+	sA, _ := RunServerWithConfig(conf)
+	defer sA.Shutdown()
+	for _, v := range []struct {
+		pass bool
+		f    func(*jwt.Limits)
+	}{
+		{true, nil},
+		{false, func(j *jwt.Limits) { j.Src = "8.8.8.8/8" }},
+		{true, func(j *jwt.Limits) { j.Src = "8.8.8.8/0" }},
+		{true, func(j *jwt.Limits) { j.Src = "127.0.0.1/8" }},
+		{true, func(j *jwt.Limits) { j.Src = "8.8.8.8/8,127.0.0.1/8" }},
+		{false, func(j *jwt.Limits) { j.Src = "8.8.8.8/8,9.9.9.9/8" }},
+		{true, func(j *jwt.Limits) { j.Times = append(j.Times, newTimeRange(time.Now(), time.Hour)) }},
+		{false, func(j *jwt.Limits) { j.Times = append(j.Times, newTimeRange(time.Now().Add(time.Hour), time.Hour)) }},
+		{true, func(j *jwt.Limits) {
+			j.Times = append(j.Times, newTimeRange(inAnHour, time.Hour), newTimeRange(time.Now(), time.Hour))
+		}}, // last one is within range
+		{false, func(j *jwt.Limits) {
+			j.Times = append(j.Times, newTimeRange(inAnHour, time.Hour), newTimeRange(inTwoHours, time.Hour))
+		}}, // out of range
+		{false, func(j *jwt.Limits) {
+			j.Times = append(j.Times, newTimeRange(inAnHour, 3*time.Hour), newTimeRange(inTwoHours, 2*time.Hour))
+		}}, // overlapping [a[]b] out of range*/
+		{false, func(j *jwt.Limits) {
+			j.Times = append(j.Times, newTimeRange(inAnHour, 3*time.Hour), newTimeRange(inTwoHours, time.Hour))
+		}}, // overlapping [a[b]] out of range
+		// next day tests where end < begin
+		{true, func(j *jwt.Limits) { j.Times = append(j.Times, newTimeRange(time.Now(), 25*time.Hour)) }},
+		{true, func(j *jwt.Limits) { j.Times = append(j.Times, newTimeRange(time.Now(), -time.Hour)) }},
+	} {
+		t.Run("", func(t *testing.T) {
+			creds := createUserWithLimit(t, kp, doNotExpire, v.f)
+			defer os.Remove(creds)
+			if c, err := nats.Connect(sA.ClientURL(), nats.UserCredentials(creds)); err == nil {
+				c.Close()
+				if !v.pass {
+					t.Fatalf("Expected failure got none")
+				}
+			} else if v.pass {
+				t.Fatalf("Expected success got %v", err)
+			} else if !strings.Contains(err.Error(), "Authorization Violation") {
+				t.Fatalf("Expected error other than %v", err)
+			}
+		})
+	}
+}
+
+func TestJWTTimeExpiration(t *testing.T) {
+	validFor := 4 * time.Second
```

Comment:
Setting this to `1500 * time.Millisecond` still passes and test goes from 19.5s down to ~6sec. Could you change?

Reply:
fixed. Also changed the test interval to be in the same ballpark than validFor